src/storage/storageController.ts

```diff
@@ -103,103 +100,96 @@ export default class StorageController {
     );
   }
 
-  saveConnectionToWorkspaceStore(
-    connection: SavedConnection
-  ): Thenable<void> {
+  async saveConnectionToWorkspaceStore(storeConnectionInfo: StoreConnectionInfo): Promise<void> {
     // Get the current save connections.
-    let workspaceConnections: StoredConnectionsType = this.get(
+    let workspaceConnections = this.get(
       StorageVariables.WORKSPACE_SAVED_CONNECTIONS,
-      StorageScope.WORKSPACE
+      StorageLocation.WORKSPACE
     );
+
     if (!workspaceConnections) {
       workspaceConnections = {};
     }
 
-    connection.storageLocation = StorageScope.WORKSPACE;
-
     // Add the new connection.
-    workspaceConnections[connection.id] = connection;
+    workspaceConnections[storeConnectionInfo.id] = storeConnectionInfo;
 
     // Update the store.
     return this.update(
       StorageVariables.WORKSPACE_SAVED_CONNECTIONS,
       workspaceConnections,
-      StorageScope.WORKSPACE
+      StorageLocation.WORKSPACE
     );
   }
 
-  storeNewConnection(newConnection: SavedConnection): Thenable<void> {
+  async saveConnection(storeConnectionInfo: StoreConnectionInfo): Promise<StoreConnectionInfo> {
     const dontShowSaveLocationPrompt = vscode.workspace
       .getConfiguration('mdb.connectionSaving')
       .get('hideOptionToChooseWhereToSaveNewConnections');
 
     if (dontShowSaveLocationPrompt === true) {
       // The user has chosen not to show the message on where to save the connection.
       // Save the connection in their default preference.
-      const preferedStorageScope = vscode.workspace
-        .getConfiguration('mdb.connectionSaving')
-        .get('defaultConnectionSavingLocation');
-
-      if (preferedStorageScope === DefaultSavingLocations.Workspace) {
-        return this.saveConnectionToWorkspaceStore(newConnection);
-      } else if (preferedStorageScope === DefaultSavingLocations.Global) {
-        return this.saveConnectionToGlobalStore(newConnection);
+      storeConnectionInfo.storageLocation = this.getPreferedStorageLocationFromConfiguration();
+    } else {
+      storeConnectionInfo.storageLocation = await this.getStorageLocationFromPrompt();
+    }
+
+    if (storeConnectionInfo.storageLocation === StorageLocation.WORKSPACE) {
+      await this.saveConnectionToWorkspaceStore(storeConnectionInfo);
+    } else if (storeConnectionInfo.storageLocation === StorageLocation.GLOBAL) {
+      await this.saveConnectionToGlobalStore(storeConnectionInfo);
+    }
```

Comment:
(non-blocking:) These two methods look very similar – would it make sense to share their code, and pass in the storage location/variable name as a parameter instead?

Reply:
Yeah, I improved it a bit already comparing to main, but you are right and can be done better...)